### PR TITLE
Update Mailgun.Client to use async/await.

### PR DESCRIPTION
**Pull request by mbrandonw**

### CODE REVIEW

File: Sources/PointFree/Account/Team.swift [hunk 1]
```diff
@@ -101,27 +99,22 @@ private let requireTeammate: MT<Tuple2<User.ID, User>, Tuple2<User, User>> = fil
...
-  }
+  async let response1 = sendEmail(
+    to: [teammate.email],
+    subject: "You have been removed from \(owner.displayName)’s Point-Free team",
+    content: inj2(youHaveBeenRemovedEmailView(.teamOwner(owner)))
+  )
 
-  return zip2(
-    parallel(
-      sendEmail(
```
Comment (mbrandonw) [inline Sources/PointFree/Account/Team.swift:112]: replacing a fire-and-forgot-zip-parallel with a few `async let`s.

File: Sources/PointFree/Account/UpdateProfile.swift [hunk 2]
```diff
@@ -60,15 +60,13 @@ private func updateProfileMiddlewareHandler(
   let updateFlash: Middleware<HeadersOpen, HeadersOpen, Prelude.Unit, Prelude.Unit>
   if data.email.rawValue.lowercased() != user.email.rawValue.lowercased() {
     updateFlash = flash(.warning, "We've sent an email to \(user.email) to confirm this change.")
-    parallel(
-      sendEmail(
+    Task {
+      _ = try await sendEmail(
```
Comment (mbrandonw) [inline Sources/PointFree/Account/UpdateProfile.swift:64]: using `Task { ... }` for fire-and-forget in a few spots.

File: Sources/PointFree/Account/Team.swift [hunk 3]
```diff
@@ -83,9 +83,7 @@ let removeTeammateMiddleware =
       try await Current.database
         .removeTeammateUserIdFromSubscriptionId(teammate.id, teammateSubscriptionId)
 
-      // Fire-and-forget emails to owner and teammate
-      sendEmailsForTeammateRemoval(owner: currentUser, teammate: teammate)
-        .run({ _ in })
+      try await sendEmailsForTeammateRemoval(owner: currentUser, teammate: teammate)
```
Comment (stephencelis) [inline Sources/PointFree/Account/Team.swift:86]: Should this be wrapped in a `Task`?

File: Sources/PointFree/Account/Team.swift (hunk 1 above)
Comment (stephencelis) [reply Sources/PointFree/Account/Team.swift:112]: I think the one difference here is that `async let`s will still throw and cause the request to fail. That might be fine, but if we wanna stay on the happy path for the request maybe we should use `Task` here as well?

**Review by stephencelis — APPROVED** [review]
Nice! Just a coupla small questions.

File: Sources/PointFree/Account/Team.swift (hunk 3 above)
Comment (mbrandonw) [reply Sources/PointFree/Account/Team.swift:86]: yep good catch. updating now.

File: Sources/PointFree/Account/Team.swift (hunk 1 above)
Comment (mbrandonw) [reply Sources/PointFree/Account/Team.swift:112]: yeah, I decided to just inline it all: https://github.com/pointfreeco/pointfreeco/pull/805/commits/95319eba064fdc4abe106583e1b5bbcb9ccff968

**Pull request merged**